packages/teams-js/src/public/meeting.ts

```diff
@@ -560,4 +601,121 @@ export namespace meeting {
+   * set isAppHandlingAudio=true in params, will set the Host(Teams) audio to go audioless
+   *   Register the mic mute status change events, and App will be receive the events.
+   * set isAppHandlingAudio=false in params, will restore the Host(Teams) audio
+   *   Deregister the mic mute status change events and App will not be receive the events.
+   *
+   * @throws standard parameter validation error
+   * @param requestAppAudioHandlingParams - RequestAppAudioHandlingParams interface, and contains
+   *   isAppHandlingAudio - true - expect App to handle the audio and Teams to go audioless.
+   *     false - Audio transferred to Teams and Teams to restore the audio.
+   *   callbackMicMuteStateChangedHandler - Callback contains mic status param.
+   * @param callback - Callback contains error and operation result parameters.
+   * error can either contain an error of type SdkError (error indication), or null (non-error indication)
+   * result can either contain a true boolean value (successful termination), or null (unsuccessful fetch)
+   *
+   * @beta
+   */
+  export function requestAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[requestAppAudioHandling] Callback response cannot be null');
+    }
+    if (!requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler) {
+      throw new Error('[requestAppAudioHandling] Callback Mic mute state handler cannot be null');
+    }
+    ensureInitialized(runtime, FrameContexts.sidePanel, FrameContexts.meetingStage);
+
+    if (requestAppAudioHandlingParams.isAppHandlingAudio) {
+      startAppAudioHandling(requestAppAudioHandlingParams, callback);
+    } else {
+      stopAppAudioHandling(requestAppAudioHandlingParams, callback);
+    }
+  }
+
+  function startAppAudioHandling(
+    requestAppAudioHandlingParams: RequestAppAudioHandlingParams,
+    callback: (isHostAudioless: boolean) => void,
+  ): void {
+    const callbackInternalRequest = (error: SdkError | null, isHostAudioless: boolean | null): void => {
+      if (error && isHostAudioless != null) {
+        throw new Error('[requestAppAudioHandling] Callback response - both parameters cannot be set');
+      }
+      if (error) {
+        throw new Error(`[requestAppAudioHandling] Callback response - SDK error ${error.errorCode} ${error.message}`);
+      }
+      if (typeof isHostAudioless !== 'boolean') {
+        throw new Error('[requestAppAudioHandling] Callback response - isHostAudioless must be a boolean');
+      }
+
+      const micStateChangedCallback = async (micState: MicState): Promise<void> => {
+        const newMicState = await requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler(micState);
+
+        const micStateDidUpdate = newMicState.isMicMuted === micState.isMicMuted;
+        setMicStateWithReason(
```

Comment:
How to send the error code to Host?

Reply:
Would it make sense to have a try catch here and then use a new MicStateChangeReason `MicStateChangeReason.AppFailedToChange`?

Reply:
Are you trying to handle cases where requestAppAudioHandlingParams.callbackMicMuteStateChangedHandler throws an error? If so you could do something like that, but you'll need to be careful about what data about the error you pass up to the host (You can't pass the message for example because it may contain information we aren't allowed to log).

Reply:
Add the change for this